### PR TITLE
Update scripts on protocol contracts

### PR DESCRIPTION
**Pull request by andresaiello**
### Summary
- Add deployment scripts
- Update references

### CODE REVIEW

**Comment by fadeev** [comment]
@andresaiello couldn't do "Suggest changes" on GH, so I pushed changes to fix imports.

**Comment by fadeev** [comment]
I'm getting the following when trying to deploy:

```
npx hardhat run ./scripts/deployments/core/deploy.ts --network athens

Getting tss address from athens: athens.
Getting tssUpdater address from athens: athens.
Error: resolver or addr is not configured for ENS name (argument="name", value="", code=INVALID_ARGUMENT, version=contracts/5.6.2)
    at Logger.makeError (/Users/fadeev/github.com/zeta-chain/protocol-contracts/node_modules/@ethersproject/logger/src.ts/index.ts:269:28)
    at Logger.throwError (/Users/fadeev/github.com/zeta-chain/protocol-contracts/node_modules/@ethersproject/logger/src.ts/index.ts:281:20)
    at Logger.throwArgumentError (/Users/fadeev/github.com/zeta-chain/protocol-contracts/node_modules/@ethersproject/logger/src.ts/index.ts:285:21)
    at /Users/fadeev/github.com/zeta-chain/protocol-contracts/node_modules/@ethersproject/contracts/src.ts/index.ts:140:16
    at step (/Users/fadeev/github.com/zeta-chain/protocol-contracts/node_modules/@ethersproject/contracts/lib/index.js:48:23)
    at Object.next (/Users/fadeev/github.com/zeta-chain/protocol-contracts/node_modules/@ethersproject/contracts/lib/index.js:29:53)
    at fulfilled (/Users/fadeev/github.com/zeta-chain/protocol-contracts/node_modules/@ethersproject/contracts/lib/index.js:20:58) {
  reason: 'resolver or addr is not configured for ENS name',
  code: 'INVALID_ARGUMENT',
  argument: 'name',
  value: ''
}
```

I'm sure it's just me doing something wrong. I'd love to document the process (and just generally the repo) to make it accessible to external contributors, but I might need some guidance.

File: scripts/deployments/core/deterministic-deploy-zeta-connector.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import { ZetaConnectorEth__factory, ZetaConnectorNonEth__factory } from "../../../typechain-types";
+import { BigNumber } from "ethers";
+import { ethers, network } from "hardhat";
+
+import { getAddress } from "../../../lib/address.helpers";
+import { ZETA_CONNECTOR_SALT_NUMBER_ETH, ZETA_CONNECTOR_SALT_NUMBER_NON_ETH } from "../../../lib/contracts.constants";
+import { isEthNetworkName } from "../../../lib/contracts.helpers";
+import { deployContractToAddress, saltToHex } from "../../../lib/ImmutableCreate2Factory/ImmutableCreate2Factory.helpers";
+
+const DEPLOYER_ADDRESS = process.env.DEPLOYER_ADDRESS ?? "";
```
Comment (fadeev) [inline scripts/deployments/core/deterministic-deploy-zeta-connector.ts:12]: Isn't `DEPLOYER_ADDRESS` deterministically derived from `PRIVATE_KEY`? Unless I'm missing something and it's a different address.

File: scripts/deployments/core/deterministic-deploy-zeta-connector.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    constructorTypes,
+    contractBytecode,
+    factoryAddress: immutableCreate2Factory,
+    salt: salthex,
+    signer
+  });
+
+  console.log("Deployed ZetaConnector. Address:", address);
+  console.log("Constructor Args", constructorArgs);
+  saveAddress("connector", address);
```
Comment (fadeev) [inline scripts/deployments/core/deterministic-deploy-zeta-connector.ts:57]: Shall we temporarily comment out `saveAddress`, because currently it fails.

Comment (andresaiello) [reply scripts/deployments/core/deterministic-deploy-zeta-connector.ts]: done

File: scripts/deployments/core/deterministic-deploy-zeta-connector.ts (hunk 1 above)
Comment (andresaiello) [reply scripts/deployments/core/deterministic-deploy-zeta-connector.ts:12]: will fallback to deployer

**Comment by fadeev** [comment]
I'm getting:

```
npx hardhat run scripts/deployments/core/deploy.ts --network polygon-mumbai

Deploying ZetaConnector to polygon-mumbai
Getting zetaToken address from athens: polygon-mumbai.
Getting tss address from athens: polygon-mumbai.
Getting tssUpdater address from athens: polygon-mumbai.
Getting tssUpdater address from athens: polygon-mumbai.
Getting tss address from athens: polygon-mumbai.
Getting tssUpdater address from athens: polygon-mumbai.
Getting tss address from athens: polygon-mumbai.
Getting tssUpdater address from athens: polygon-mumbai.
Getting zetaToken address from athens: polygon-mumbai.
Updating
Getting tss address from athens: polygon-mumbai.
Getting connector address from athens: polygon-mumbai.
ProviderError: INTERNAL_ERROR: could not replace existing tx
    at HttpProvider.request (/Users/fadeev/github.com/zeta-chain/protocol-contracts/node_modules/hardhat/src/internal/core/providers/http.ts:88:21)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async EthersProviderWrapper.send (/Users/fadeev/github.com/zeta-chain/protocol-contracts/node_modules/@nomiclabs/hardhat-ethers/src/internal/ethers-provider-wrapper.ts:13:20)
```

**Review by fadeev — DISMISSED** [review]
👍

**Pull request merged**